### PR TITLE
Dev akash sysloglib

### PR DESCRIPTION
**Pull request by akashkollipara**

### CODE REVIEW

**Comment by pranjalchanda08** [comment]
> Kudos, SonarCloud Quality Gate passed!    Quality Gate passed
> 
> [Bug](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=CODE_SMELL) [8 Code Smells](https://sonarcloud.io/project/issues?id=VisorFolks_cyancore&pullRequest=107&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=VisorFolks_cyancore&pullRequest=107&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=VisorFolks_cyancore&pullRequest=107&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=VisorFolks_cyancore&pullRequest=107&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=VisorFolks_cyancore&pullRequest=107&metric=new_duplicated_lines_density&view=list)

@akashkollipara  Can you check for the code-smells?

**Comment by akashkollipara** [comment]
Yes, will do... but many don't make sense because of variable args. can we disable that?

File: src/lib/libc/printf.c [hunk 1]
```diff
@@ -175,7 +200,27 @@ int printf(const char *fmt, ...)
 	int ret;
 	va_list va;
 	va_start(va, fmt);
-	ret = vprintf(fmt, va);
+	ret = vprintf(stdout, 0, fmt, va);
+	va_end(va);
+	return ret;
+}
+
+int eprintf(const char *fmt, ...)
```
Comment (pranjalchanda08) [inline src/lib/libc/printf.c:208]: make this as **perror** as standard with libc

File: src/lib/libsyslog/syslog.c [hunk 2]
```diff
@@ -1,304 +1,54 @@
...
-static syslog_ctrl_t syslog_ctrl;
-
-/**
- * syslog_api - Placeholder pointing to all the API related to syslog
- *
- * This variable is used to contain the pointers to syslog APIs to be used by the user.
- * This variable is only accesable using syslog interface i.e. g_syslog
- */
-static syslog_api_t syslog_api =
+int syslog(logtype_t t, const char *c, ...)
```
Comment (pranjalchanda08) [inline src/lib/libsyslog/syslog.c:26]: Add documentation comments

**Review by pranjalchanda08 — CHANGES_REQUESTED** [review]

File: src/lib/libsyslog/syslog.c (hunk 2 above)
Comment (akashkollipara) [reply src/lib/libsyslog/syslog.c:26]: I will add later...

File: src/lib/libc/printf.c (hunk 1 above)
Comment (akashkollipara) [reply src/lib/libc/printf.c:208]: perror works differently... so intentionally kept it like this.

**Review by pranjalchanda08 — APPROVED** [review]

**Pull request merged**